tests/mpc.py

```diff
@@ -0,0 +1,240 @@
+
+    def impl_gradient(self, result, x, functionId):
+        raise NotImplementedError
+
+    def impl_jacobian(self, result, x):
+        AJ, VJ, PJ = compute_jacs(self.params)
+        result[:,:] = PJ
+
+
+class TestMPCPy(unittest.TestCase):
+
+    def test_mpc(self):
+        """
+        Generate speed profile by integrating jerk
+        This test provide big constraint jacobian matrix
+        that make him fail if numpy matrix and roboptim internal matrix stride mismatch
+        """
+        #                      N,  dt,  a0, v0, p0, x_target
+        params = ProblemParams(30, 0.2, 0., 0., 0., 5.)
+        max_jerk = 0.6
+
+        # J vector used by finite difference test
+        TJ = np.random.rand(params.N)
+
+
+        # Cost
+        jerk_norm = JerkNorm(params)
+        jerk_norm_finite = roboptim.core.PyFiniteDifference(jerk_norm)
+
+        pos_target = PosTarget(params)
+        pos_target_finite = roboptim.core.PyFiniteDifference(pos_target)
+
+        cost = Cost(params, [(0.1, jerk_norm), (1., pos_target)])
+        cost_finite = roboptim.core.PyFiniteDifference(cost)
+
+
+        # Constraints
+        vel = Velocity(params)
+        vel_finite = roboptim.core.PyFiniteDifference(vel)
+
+        last_vel = LastVelocity(params)
+        last_vel_finite = roboptim.core.PyFiniteDifference(last_vel)
+
+        pos = Position(params)
+        pos_finite = roboptim.core.PyFiniteDifference(pos)
+
+        # Validate jacobian
+        np.testing.assert_almost_equal(jerk_norm.jacobian(TJ), jerk_norm_finite.jacobian(TJ), 1e-4)
+        np.testing.assert_almost_equal(pos_target.jacobian(TJ), pos_target_finite.jacobian(TJ), 1e-4)
+        np.testing.assert_almost_equal(cost.jacobian(TJ), cost_finite.jacobian(TJ), 1e-4)
+        np.testing.assert_almost_equal(vel.jacobian(TJ), vel_finite.jacobian(TJ), 1e-4)
+        np.testing.assert_almost_equal(last_vel.jacobian(TJ), last_vel_finite.jacobian(TJ), 1e-4)
+        np.testing.assert_almost_equal(pos.jacobian(TJ), pos_finite.jacobian(TJ), 1e-4)
+
+        # Create problem
+        problem = roboptim.core.PyProblem(cost)
+        problem.startingPoint = np.array([0.]*params.N)
+        problem.argumentBounds = np.array([(-max_jerk, max_jerk)]*params.N)
+        problem.addConstraint(pos, np.array([(0., params.x_target)]*params.N))
+        problem.addConstraint(last_vel, np.array([(3., 3.)]))
+        problem.addConstraint(vel, np.array([(0., 10.)]*params.N))
+
+
+        solver = roboptim.core.PySolver("ipopt", problem)
+        solver.setParameter("ipopt.output_file", "mpc.log")
+        solver.setParameter("ipopt.hessian_approximation", "limited-memory")
+        # We can check with the ipopt derivative checker that previously checked jacobian
+        # are not well forwarded to ipopt
+        solver.setParameter("ipopt.derivative_test", 'first-order')
+
+        solver.solve()
+        r = solver.minimum()
+
+        self.assertTrue(isinstance(r, roboptim.core.PyResult))
```

Comment:
Does this problem have a global optimum that we could check?

Reply:
Since objective is quadratic and constraints linear a global optimum exist and is easy to find.